src/libfetchers/git.cc

```diff
@@ -714,6 +729,11 @@ struct GitInputScheme : InputScheme
 
         auto repoInfo = getRepoInfo(input);
 
+        if (getExportIgnoreAttr(input)
+            && getSubmodulesAttr(input)) {
+            throw UnimplementedError("exportIgnore and submodules are not supported together yet");
```

Comment:
Note that this combination isn't used in any existing expressions, because the previous implementation did not apply export-ignore when submodules were enabled.

Reply:
Apart from this check, why don't they work together yet? Given that submodules are also implemented using Git accessors, I would expect the filtering to just work for submodules.

Reply:
I've added the comment:
```
            /* In this situation, we don't have a git CLI behavior that we can copy.
               `git archive` does not support submodules, so it is unclear whether
               rules from the parent should affect the submodule or not.
               When git may eventually implement this, we need Nix to match its
               behavior. */
```